src/main/java/com/dreammaster/gthandler/recipes/AlloySmelterRecipes.java

```diff
@@ -8,236 +11,343 @@
 import com.dreammaster.gthandler.CustomItemList;
 
 import cpw.mods.fml.common.Loader;
-import gregtech.api.enums.GT_Values;
-import gregtech.api.enums.ItemList;
-import gregtech.api.enums.Materials;
-import gregtech.api.enums.OrePrefixes;
+import gregtech.api.enums.*;
 import gregtech.api.util.GT_ModHandler;
 import gregtech.api.util.GT_OreDictUnificator;
 import gregtech.api.util.GT_Utility;
 
 public class AlloySmelterRecipes implements Runnable {
 
+    final String modIDEnderIO = "EnderIO";
+    final String modIDTinkerConstruct = "TConstruct";
+
     @Override
     public void run() {
+        registerCoreModRecipes();
+        registerEnderIORecipes();
+        registerTinkerConstructRecipes();
+
+    }
+
+    public void registerTinkerConstructRecipes() {
+        if (!Loader.isModLoaded(modIDTinkerConstruct)) {
```

Comment:
If mod ids are in an enum, the enum could have a method for this like `Mods.TinkersConstruct.isLoaded()`

Reply:
The `isLoaded` status could be cached to save the costly repeated calls to `Loader.isModLoaded`. Gregtech already caches some of those.

Reply:
I added a new enum for that, with some lazy cache population.